libs/barista-components/drawer-table/src/drawer-table.ts

```diff
@@ -0,0 +1,188 @@
+})
+export class DtDrawerTable<T> implements AfterViewInit, AfterContentInit {
+  /** Columns that stay visible when row is expanded  */
+  @Input() openColumns: string[];
+
+  /** Reference to the drawer element  */
+  @ViewChild(DtDrawer, { static: true }) private drawer: DtDrawer;
+
+  /** Reference to the drawer container  */
+  @ViewChild('container', { read: ViewContainerRef, static: false })
+  private container: ViewContainerRef;
+
+  /** @internal Template reference for the drawer content. */
+  @ContentChild(TemplateRef, { static: false })
+  private _drawerContentRef: TemplateRef<{
+    $implicit: T;
+  }>;
+
+  /** @internal Reference to the table that is used to make the rows interactive */
+  @ContentChild(DtTable, { static: false }) private _table: DtTable<T>;
+
+  /** @internal Reference to the header row definition which is used to set the columns  */
+  @ContentChild(DtHeaderRowDef, { static: false })
+  private _rowHeaderDef: DtHeaderRowDef;
+
+  /** Reference to the row definition which is used to get and set the columns  */
+  @ContentChild(DtRowDef, { static: false }) rowDef: DtRowDef<T>;
+  private initialColumns: Iterable<string>;
+
+  /** @internal Used to track the currently clicked row */
+  private _currentRow: T;
+
+  constructor(private _changeDetectorRef: ChangeDetectorRef) {}
+
+  get currentRow(): T {
+    return this._currentRow;
+  }
+
+  ngAfterContentInit(): void {
+    this._table.interactiveRows = true;
+  }
+
+  ngAfterViewInit(): void {
+    this.initialColumns = this.rowDef.columns;
+  }
+
+  /** Whether the drawer is open  */
+  isOpen(): boolean {
+    return !isEmpty(this.drawer) && this.drawer.opened;
+  }
+
+  /** Sets the drawer content */
+  setDrawerContent(row: T): void {
+    this.container.clear();
+    this.container.createEmbeddedView(this._drawerContentRef, {
+      $implicit: row,
+    });
+  }
+
+  /** @internal Row click opens the drawer and sets its content or closes if the drawer is already open */
+  _onRowClicked(row: T): void {
+    const open = !this.drawer.opened || this._currentRow !== row;
+    if (open) {
+      this.open();
+      this.setDrawerContent(row);
+    } else {
+      this.close();
+    }
+    this._currentRow = row;
+  }
+
+  getOpenColumns(): string[] {
+    return isEmpty(this.openColumns)
+      ? [this.initialColumns[0]]
+      : this.openColumns;
+  }
+
+  /** @internal Sets the columns of the table */
+  private _setColumns(columns: Iterable<string>): void {
+    this._rowHeaderDef.columns = columns;
+    this.rowDef.columns = columns;
+  }
+
+  /**
+   * Opens the drawer if it is not already opened.
+   */
+  open(): void {
+    this._changeDetectorRef.markForCheck();
+    this.drawer.open();
+    this._setColumns(this.getOpenColumns());
+  }
+
+  /**
+   * Closes the drawer if it is not already closed.
+   */
+  close(): void {
+    this._changeDetectorRef.markForCheck();
+    this.drawer.close();
+    this._setColumns(this.initialColumns);
```

Comment:
Although it should not matter too much, but maybe call `markForCheck` after you have done the changes: 
```suggestion
    this.drawer.close();
    this._setColumns(this.initialColumns);
    this._changeDetectorRef.markForCheck();
```